src/software/ai/evaluation/keep_away.cpp

```diff
@@ -0,0 +1,100 @@
+#include "software/ai/evaluation/keep_away.h"
+
+#include "software/ai/passing/cost_function.h"
+#include "software/geom/algorithms/closest_point.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/math/math_functions.h"
+#include "software/optimization/gradient_descent_optimizer.h"
+
+double ratePasserPointForKeepAway(const Pass& pass, const Team& enemy_team)
+{
+    // the default values for these passing parameters
+    // TODO: cleanup passing parameters as part of #1987
+    static constexpr double PASSER_ENEMY_PROXIMITY_IMPORTANCE   = 1.5;
+    static constexpr double RECEIVER_ENEMY_PROXIMITY_IMPORTANCE = 0.;
+    static const auto ENEMY_REACTION_TIME = Duration::fromSeconds(0);
+
+
+    return ratePassEnemyRisk(enemy_team, pass, ENEMY_REACTION_TIME,
+                             RECEIVER_ENEMY_PROXIMITY_IMPORTANCE) *
+           (1 - std::max(0., calculateProximityRisk(pass.passerPoint(), enemy_team,
+                                                    PASSER_ENEMY_PROXIMITY_IMPORTANCE)));
+}
+
+
+Point findKeepAwayTargetPoint(const World& world, const Pass& best_pass_so_far)
+{
+    static constexpr auto KEEPAWAY_SEARCH_CIRCLE_RADIUS = 0.5;
+
+    // the width of both the field boundary sigmoid and the circular search region sigmoid
+    static constexpr auto SIGMOID_WIDTH = 0.1;
+    // the inward offset of the field boundaries to use for the field lines sigmoid
+    static constexpr auto FIELD_SIZE_REDUCTION_M = 0.25;
+
+    static constexpr auto GRADIENT_STEPS_PER_ITER = 5;
```

Comment:
wondering if any constants in this file need to scale with div a field

Reply:
No, this would only need to increase if we wanted to search a larger area, which would be motivated by higher acceleration/speed limit